app/web/features/search/SearchPage.tsx

```diff
@@ -52,193 +55,195 @@ const useStyles = makeStyles((theme) => ({
+    isLoading: isLoadingUser,
+  } = useCurrentUser();
   const { t } = useTranslation([GLOBAL, SEARCH]);
+  const queryClient = new QueryClient();
   const classes = useStyles();
   const theme = useTheme();
-
   const map = useRef<MaplibreMap>();
+
+  // State
+  const [locationResult, setLocationResult] = useState({
+    bbox: bbox,
+    isRegion: false,
+    location: { lng: undefined, lat: undefined },
+    name: locationName,
+    simplifiedName: locationName,
+  });
+  const [queryName, setQueryName] = useState<undefined | string>(undefined);
+  const [searchType, setSearchType] = useState("location");
+  const [lastActiveFilter, setLastActiveFilter] = useState(0);
+  const [hostingStatusFilter, setHostingStatusFilter] = useState(0);
+  const [numberOfGuestFilter, setNumberOfGuestFilter] = useState(undefined);
+  const [completeProfileFilter, setCompleteProfileFilter] = useState(true);
   const [selectedResult, setSelectedResult] = useState<
     Pick<User.AsObject, "username" | "userId" | "lng" | "lat"> | undefined
   >(undefined);
-
-  const previousResult = usePrevious(selectedResult);
-
-  const [areClustersLoaded, setAreClustersLoaded] = useState(false);
-
-  const showResults = useRef(false);
-
-  const searchFilters = useRouteWithSearchFilters(searchRoute);
-
-  const updateMapBoundingBox = (
-    newBindingBox: [number, number, number, number] | undefined
-  ) => {
-    if (newBindingBox && newBindingBox.join() !== "0,0,0,0") {
-      map.current?.fitBounds(newBindingBox);
-    }
-  };
-
-  useEffect(() => {
-    if (showResults.current !== searchFilters.any) {
-      showResults.current = searchFilters.any;
-      setTimeout(() => {
-        map.current?.resize();
-      }, theme.transitions.duration.standard);
-    }
-  }, [searchFilters.any, selectedResult, theme.transitions.duration.standard]);
-
-  const flyToUser = useCallback((user: Pick<User.AsObject, "lng" | "lat">) => {
-    map.current?.stop();
-    map.current?.easeTo({
-      center: [user.lng, user.lat],
-    });
-  }, []);
-
-  useEffect(() => {
-    if (previousResult) {
-      //unset the old feature selection on the map for styling
-      areClustersLoaded &&
-        map.current?.setFeatureState(
-          { source: "clustered-users", id: previousResult.userId },
-          { selected: false }
+  const [isFiltersOpen, setIsFiltersOpen] = useState(false);
+  const [mapInitiallyLocated, setMapInitiallyLocated] = useState(false);
+
+  // Loads the list of users
+  const { data, error, isLoading, fetchNextPage, isFetching, hasNextPage } =
+    useInfiniteQuery<UserSearchRes.AsObject, Error>(
+      [
+        "userSearch",
+        queryName,
+        locationResult?.name,
+        locationResult?.bbox,
+        lastActiveFilter,
+        hostingStatusFilter,
+        numberOfGuestFilter,
+        completeProfileFilter,
+      ],
+      ({ pageParam }) => {
+        const lastActiveComparation = parseInt(
+          lastActiveFilter as unknown as string
```

Comment:
Is it really `unknown` or is there a type you could make for this? Since it's stored in the state I'm assuming the data is predictable enough to make a type?